### PR TITLE
Free some disk space from runner and revert static dotnet change

### PR DESCRIPTION
**Pull request by kamilchodola**
THIS WORKS - but not proud of it.

Removes some not needed build packages to fit again into smallest runners (avoid spending extra $ on build action and allows still to use latest dotnet).

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [X] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [X] Yes
- [ ] No

### CODE REVIEW

**Review by LukaszRozmej — APPROVED** [review]

**Comment by kamilchodola** [comment]
@LukaszRozmej this is just an experiment so no merge please

**Review by LukaszRozmej — APPROVED** [review]

File: .github/workflows/build-solutions.yml [hunk 1]
```diff
@@ -16,6 +16,16 @@ jobs:
         config: [release, debug]
         solution: [Nethermind, EthereumTests, Benchmarks]
     steps:
+      - name: Free Disk Space
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+          android: true
+          dotnet: false
+          haskell: true
```
[the remark this answers is not in the document]
Comment (kamilchodola) [reply .github/workflows/build-solutions.yml:26]: Runners have preinstalled libraries - this just simply removes them to reduce used disk space.
This PR is still to be closed as probably the problem we have is a dotnet issue with performance of build

**Comment by emlautarom1** [comment]
@rubo Do we need this PR?

**Comment by rubo** [comment]
@emlautarom1 Nope, a similar solution has already been applied.

**Comment by emlautarom1** [comment]
Closing since a similar solutions has already been applied.

**Pull request closed without merging**